src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.Unix.cs

```diff
@@ -60,12 +64,23 @@ public sealed partial class TarWriter : IDisposable
 
             entry._header._mode = (status.Mode & 4095); // First 12 bits
 
-            entry.Uid = (int)status.Uid;
-            entry.Gid = (int)status.Gid;
+            // Uid and UName
+            entry._header._uid = (int)status.Uid;
+            if (!_userIdentifiers.TryGetValue(status.Uid, out string? uName))
+            {
+                uName = Interop.Sys.GetUName(status.Uid);
+                _userIdentifiers.Add(status.Uid, uName);
```

Comment:
Just to be sure, instance methods of TarWriter are not thread-safe, correct? We don't allow multiple threads to be using TarWriter at the same time.

Reply:
It's not thread-safe as it is right now. I could change this to `ConcurrentDictionary` but there are other places that we would have to adjust to make sure the code is thread-safe.

Reply:
Note - I’m not asking for TarWriter to be thread safe. I don’t think it should be, unless we have some important scenario for it. I’m just checking that we don’t have those requirements.

Reply:
Good to know. Thanks for clarifying.